src/common/withPicker.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+
+import type { KeyExtractor, PickerValue } from '../stores/PickerStore';
+
+import * as React from 'react';
+import PickerStore from '../stores/PickerStore';
+import { observer } from 'mobx-react';
+import hoistNonReactStatic from 'hoist-non-react-statics';
+
+export type PickerProps<TEntity> = {
+  checkIsSelected: (item: TEntity) => boolean,
+  clear: () => void,
+  stringValue: string,
+  toggleItem: (item: TEntity) => void,
+  value: PickerValue<TEntity>,
+};
+
+type Props<TEntity> = {
+  keyExtractor?: KeyExtractor<TEntity>,
+  multiple?: boolean,
+  onChange?: (value: PickerValue<TEntity>) => void,
+  value?: PickerValue<TEntity>,
+};
+
+const withPicker = <TEntity, TProps: Props<TEntity>>(
+  Component: React.ComponentType<TProps>,
+): React.ComponentType<TProps & PickerProps<TEntity>> => {
+  @observer
+  class WithPicker extends React.Component<TProps & PickerProps<TEntity>> {
+    _pickerStore: PickerStore<TEntity>;
+
+    componentWillMount() {
```

Comment:
Somewhere I read that the `componentWill*****` functions are going away in newer react. Maybe move this stuff to a constructor.

Reply:
hm...I rather wait when it will be actually deprecated. Maybe they'll introduce something new(besides constructor) for that.

Reply:
Nope, they are just getting rid of them. They are going to move to async rendering and the `Will` functions don't work very well. The `componentDid***` are still sticking around.